six/modules/c++/six.sidd/include/six/sidd/SFA.h

```diff
@@ -205,8 +201,10 @@ struct SFAPolyhedralSurface : public SFASurface
         SFASurface(TYPE_NAME)
     {
     }
-    virtual ~SFAPolyhedralSurface()
+
+    virtual SFAPolyhedralSurface* clone() const
     {
+        return new SFAPolyhedralSurface(*this);
     }
 
     std::vector<mem::ScopedCopyablePtr<SFAPolygon> > patches;
```

Comment:
Alright, it’s changed.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Friday, February 12, 2016 3:05 PM
To: ngageoint/six-library <six-library@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>
Subject: Re: [six-library] Fix object slicing issue with Annotations, clean up some warnings (#48)


In six/modules/c++/six.sidd/include/six/sidd/SFA.h<http://cp.mcafee.com/d/1jWVIgdEIfLIT7fEK6XCSm6hMUQsLLCTNPb3a9EVd79J6XRTD3hPsQsL6QkmkkTPhOUM-r4qMSpfosH-ME2BkdrFOVJl3mWsKr8eWPV1N_HY-yCOOMDRXBQXLIYztNdUQsTuWyaqRQRrLIYG7DR8OJMddECQPt-hojuv78I9CzATsSjDdqymohBcCSgVv84t8_mk9VigunfAgzIiSbdDDUMY1y4GmrFilo2InQNNEVhspudMH7RzOMa8OCjr8sKrovjodwLQzh0qmWi3OVYy4tymdbFEw5V1VEwGg4vJcQgkfBqk29EwBizvCy0NapcQg248lDIVlxg0Ph1o_Zwq86Rzh1iGLCy0NIoiwhEq83hg1z3NEFdL6zAS9hCXl9otn3c>:

>      {

> +        return new SFAPolyhedralSurface(*this);

>      }

>

>      std::vector<mem::ScopedCopyablePtr<SFAPolygon> > patches;

Hmm, so I think you're right. I'd have to go dig through some C++ documentation to say for sure. But just for code clarity, let's say any time something is a base class where things are inheriting from it, let's use a vector of cloneables. I think line 285 is the only remaining one that needs to get changed.

—
Reply to this email directly or view it on GitHub<http://cp.mcafee.com/d/2DRPow838O86Qm7TSrzDQn3tPrb38UsqenTPrUVBxB4QsCzASztWXPxEVKqenzqabaarVEVsovdydorcDIel_ok1iG6JQVsSGxHtendA7tpYwU_R-vhjppojWZOWtTSuhKUCYqerLth5dqWqJTSul3PWApmU6CQPqpK_8I9LfzAm4PhOrKr9PCJhbc8OCjr8sLA2eAvHa4YF8fbDO8hS9r5CPPYouNSh_Q0ObWoUQsEKcL6UlzWNVo54pj9JAendIfFI6MnWhEwdbt91Vs-h2eNb6BQQg2YwYQgl82fSCq8a7OJa14QgiFhLPh0oBcCq8124aPSsGME0pEwIv-Md43qNEwFlnPh0oSc9g8Qd41EE0NxUQkCTzhOrFymi>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.